### ISSUE TITLE
[BUG] MK3.5 6.0.3 Firmware - Print not centered on bed.

### ISSUE THREAD

**Issue by artistassembly** [prusa3d/Prusa-Firmware-Buddy#4059]
### Printer model

MK3.5

### Firmware version

6.0.3

### Upgrades and modifications

Silicone Mod on Bed

### Printing from...

USB Drive

### Describe the bug

A print which is centered on the bed in Prusa Slicer 2.7.4 is offset to the right & not centered on the print sheet.  This can cause prints to fail as the printer attempts to extrude off the right side of the print sheet without any warning from the slicer.  The print also appears to be shifted to the front of the bed a similar amount, but there is enough clearance front to back to not cause a print failure.

I suspect this misalignment is also present in the application of the bed leveling mesh causing the compensation for the distance from the print sheet to the nozzle to be shifted as well. 

The print sheet is aligned accurately on the bed with the rear of the sheet against the 2 posts on the bed and the left & right sides of the sheet even with the sides of the bed.  This distance of the white "print area lines on this sheet are exactly 250mm x 210mm.

This issue was not present on the MK3S+ before the MK3.5 upgrade.  I'm seeing other users report this issue with their MK3.5 printers on Reddit as well as in this bug report https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3851.  

[MK3.5 Bed Offset Test.zip](https://github.com/user-attachments/files/16006538/MK3.5.Bed.Offset.Test.zip)

Overview 
![Overview 1](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/1ac5b3dc-bdbf-4670-ba96-b65c953c60fc)

Right Side
![Right Measured](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/d368771d-e19d-4e82-8cdf-70048c2de274)
![Right Rear 1](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/fe72fa9d-5559-4f18-9ee1-012beedde2be)

Left Side
![Left Measured](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/2505f235-3f1b-427c-a889-3c015a8eb0c2)
![Left Rear 1](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/c12e82e2-d04c-4c60-9912-a19e05482eef)

Front
![Left Front 1](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/7e4d580d-d108-41df-99dc-9bb5abb6f01d)
![Front Measured](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/2bf85105-9ebd-4397-b93f-27532dc12ce7)

Rear
![Rear Measured](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/13263646/464d2f34-e714-4c67-92a4-90da5406df7d)

### How to reproduce

I have attached a zip file with a test stl which measures 249mm x 209mm as well as the 3mf & .bgcode files.  Printing this model with the stock Prusa Slicer 2.7.4 MK3.5 Printer settings should produce the same result.

### Expected behavior

I would expect the print to be centered on the print sheet with greater accuracy.

**Comment by BlueFyre** [comment]
Thanks for filing this. Would like to see this acknowledged/addressed as well

**Comment by MarkusReichart** [comment]
I also have this issue. While upgrading to MK3.5 I changed from nylon lock to the new spacers for the heat bed. 
Whether the old y-carriage in combination with new spacers cause this shift, or the new homing sequence set a different 0,0 position I don`t know. In #3851 they just limit the right boundaray -5mm for PINDA. Maybe it would be better to shift the whole origin more to the left.
Meantime I define offsets in PS:

![image](https://github.com/prusa3d/Prusa-Firmware-Buddy/assets/52602624/7d7ab6ff-a83c-4eb5-b895-0c03c6ac2a73)

**Comment by nickduncan7** [comment]
I understand why Prusa says it should not be perfectly centered in the prior issue, and frankly that is not my expectation. However, if it was closer to center on the build plate, I believe the PINDA probing off the edge issue would have likely never been a problem.

**Comment by Prusa-Support** [comment]
Thanks for reporting. The shifted MBL probing grid and the shifted print area are not necessarily related so the issues should ideally be listed separately.

This should not prevent users from printing large prints and the little tweak via Prusa Slicer printer profile suggested above is surely a valid option for the time being (optionally). However, we would appreciate seeing your measured offset so that we may eventually make more considerations on this supported by numbers - the more the tested printer, the better of course. 🙂 

Michele Moramarco
Prusa Research

**Comment by nickduncan7** [comment]
> Thanks for reporting. The shifted MBL probing grid and the shifted print area are not necessarily related so the issues should ideally be listed separately.

The first sentence of the report in BFW-5606 is “The print area on the MK3.5 is not centered in the indicated print area on Prusa steel sheets.”

The first sentence of this report is “A print which is centered on the bed in Prusa Slicer 2.7.4 is offset to the right & not centered on the print sheet.”

I understand these are tracked as different causes internally, but from an external perspective the same issue has been reported twice and is ultimately still not resolved. The PINDA probing issue was mentioned solely as a side effect from the prior problem.

**Comment by timur-tabi** [comment]
In the Bed Shape settings, what number should I put for the X Origin?

**Comment by timur-tabi** [comment]
It's not stale, Prusa developers are just not interested in fixing their bugs. 

This didn't happen on the MK3.

**Comment by timur-tabi** [comment]
@Prusa-Support  Any change of this being fixed any time soon?  I cannot print a 250mm-wide object because the extruder head keeps crashing into the right side.

**Comment by Prusa-Support** [comment]
To my understanding, 250x210 mm prints were possible even if not centered.
I must be missing something but please note that an acceptable skew up to 0.25° may reduce the maximum axis length by up to 1mm or so.
https://help.prusa3d.com/article/xyz-calibration-details_2272
However, as of now, I can't recall or check if the exact same XYZ Calibration typical of MK3 is ported to MK3.5, including skew measurement.

I will report and review with the developers.
Nonetheless, more measurements and tests across more MK3.5 units may help.

Michele Moramarco
Prusa Research

**Comment by ttabi** [comment]
Fair enough.  My only suggestion is that it should be possible to specify X/Y offsets in the Bed Shape setting and still see the Prusa bed image in the Plater panel.  As soon as I specify a non-zero offset, that image is replaced with a generic rectangle.
